guide/popular-topics/intents.md

```diff
@@ -0,0 +1,104 @@
+# Gateway Intents
+
+Gateway Intents were introduced to the library in v12 and allow you to pick-and-choose which events you wish your bot to receive and/or omit.
+Intents are groups of pre-defined events that the discord.js client will conditionally subscribe to. For example, omitting the `GUILD_PRESENCES` intent would prevent the discord.js client from receiving any member-related presence events. Before this feature, the discord.js client would receive all events, thus consuming memory for data you may not be using.
+
+:::warning
+Intents are optional on the v6 gateway but will become mandatory in a future gateway version.
+:::
+
+<branch version="11.x">
+
+Intents are not available in version 11, please update to version 12 of the library if you want to use gateway intents in your bot.
+
+</branch>
+
+<branch version="12.x">
+
+## Enabling Intents
+
+You can choose which intents you'd like to receive as client options when instantiating your bot client.
+
+Permissable gateway intents include:
+- `GUILDS` - Events from this intent include:
+  - `guildCreate`
+  - `guildDelete`
+  - `roleCreate`
+  - `roleUpdate`
+  - `roleDelete`
+  - `channelCreate` (within guilds)
+  - `channelUpdate` (within guilds)
+  - `channelDelete` (within guilds)
+  - `channelPinsUpdate` (within guilds)
+- `GUILD_MEMBERS`¹ - Events from this intent include:
+  - `guildMemberAdd`
+  - `guildMemberUpdate`
+  - `guildMemberRemove`
+- `GUILD_BANS` - Events from this intent include:
+  - `guildBanAdd`
+  - `guildBanRemove`
+- `GUILD_EMOJIS` - Events from this intent include:
+  - `emojiUpdate`
+- `GUILD_INTEGRATIONS` - Events from this intent include:
+  - `guildIntegrationsUpdate`
+- `GUILD_WEBHOOKS` - Events from this intent include:
+  - `webhookUpdate`
+- `GUILD_INVITES` - Events from this intent include:
+  - `inviteCreate`
+  - `inviteDelete`
+- `GUILD_VOICE_STATES` - Events from this intent include:
+  - `voiceStateUpdate`
+- `GUILD_PRESENCES`¹ - Events from this intent include (within guilds):
+  - `presenceUpdate`
+- `GUILD_MESSAGES` - Events from this intent include:
+  - `message`
+  - `messageUpdate`
+  - `messageDelete`
+- `GUILD_MESSAGE_REACTIONS` - Events from this intent include (within guilds):
+  - `messageReactionAdd`
+  - `messageReactionRemove`
+  - `messageReactionRemoveAll`
+  - `messageActionRemoveEmoji`
+- `GUILD_MESSAGE_TYPING` - Events from this intent include (within guilds):
+  - `typingStart`
+- `DIRECT_MESSAGES` - Events from this intent include (within DMs):
+  - `channelCreate`
+  - `message`
+  - `channelUpdate`
+  - `channelDelete`
+  - `channelPinsUpdate`
+- `DIRECT_MESSAGE_REACTIONS` - Events from this intent include (within DMs):
+  - `messageReactionAdd`
+  - `messageReactionRemove`
+  - `messageReactionRemoveAll`
+  - `messageActionRemoveEmoji`
+- `DIRECT_MESSAGE_TYPING` - Events from this intent include (within DMs):
+  - `typingStart`
```

Comment:
This should not be documented in the guide, as it is directly taken from  <https://discordapp.com/developers/docs/topics/gateway#list-of-intents> it might as well just link there and explain what the headings mean and how to use them on an example